### PR TITLE
Support for extending HTTP and socket protocol

### PR DESCRIPTION
**Pull request by NadirRoGue**
Adding handlers to implement custom http requests and webstreamer event handlers on applications that make use of the webstreamer

### CODE REVIEW

File: client/src/events.ts [hunk 1]
```diff
@@ -175,6 +187,33 @@ export function serializeEvent(event: Event) {
             return buffer;
         }
 
+				case EventType.ClipboardInput:
```
Comment (soehrl) [inline client/src/events.ts]: The formatting here seems a little odd, too much indentation?

File: client/src/key-mapper.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+				}
+				else
+				{
+			
+					KeyMapper.lastKeyPressed = '';
+					return key;
+				}
+			}
+		}
+*/
```
Comment (soehrl) [inline client/src/key-mapper.ts]: Were there any problems when using combos?

File: webstreamer/CMakeLists.txt [hunk 3]
```diff
@@ -78,7 +78,7 @@ target_compile_definitions(webstreamer PUBLIC ${FFMPEG_DEFINITIONS})
 # add_definitions(${LIBSOURCEY_DEFINITIONS})
 
 # # WebRTC
-find_package(WebRTC)
+#find_package(WebRTC)
```
Comment (soehrl) [inline webstreamer/CMakeLists.txt]: Why is this commented out?

File: client/src/key-mapper.ts (hunk 2 above)
Comment (NadirRoGue) [reply client/src/key-mapper.ts]: Actually, the combo handling is commented out, this class is used mostly to translate key input from any keyboard layout into a general table that any server will understand. I kept the combo code (again, commented out) just for future reference.

File: client/src/events.ts (hunk 1 above)
Comment (NadirRoGue) [reply client/src/events.ts]: You are right, Ill check it

File: webstreamer/CMakeLists.txt (hunk 3 above)
Comment (NadirRoGue) [reply webstreamer/CMakeLists.txt]: My bad, it got into the pull request by mistake, but theres actually a problem with this library. Its a static library which includes all the OpenSSL symbols, and was creating conflicts when compiling using Poco NetSSL for other purpose needed by the remote rendering backend. So for testing purposes is commented. The goal is to replace the static library by a shared one so these problems wont arise.

Comment (NadirRoGue) [reply webstreamer/CMakeLists.txt]: I reverted the change and enabled again the find_package command.

File: client/src/events.ts (hunk 1 above)
Comment (NadirRoGue) [reply client/src/events.ts]: I fixed it (as well as other identation typos I found in the same file)

File: webstreamer/CMakeLists.txt (hunk 3 above)
Comment (soehrl) [reply webstreamer/CMakeLists.txt]: Oh okay, then we should probably switch to the shared version soon to avoid these kind of conflicts :)

**Comment by ptoharia** [comment]
@NadirRoGue @soehrl are we ready then to merge this?

**Comment by NadirRoGue** [comment]
@ptoharia It's fine for me

**Pull request merged**